### PR TITLE
Add maintenance.md for supported Kubernetes upgrade

### PR DESCRIPTION
**Pull request by tapih**
We need some special knowledge for upgrading the supported Kubernetes versions.
This PR adds a document to decrease the knowledge gap between maintainers about how to upgrade the versions.
Signed-off-by: H.Muraoka <hiroshi-muraoka@cybozu.co.jp>

### CODE REVIEW

File: docs/maintenance.md
Comment (ymmt2005) [inline docs/maintenance.md:7]:
```suggestion
## Check release notes
```

Please adjust other titles as well.

File: docs/maintenance.md
Comment (ymmt2005) [inline docs/maintenance.md:57]: To update **CSI** version, we need to update `CSI_VERSION` in Makefile, and run the following commands:

```console
$ make csi.proto
$ make generate 
```

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Review by toshipp — DISMISSED** [review]
LGTM.
I'm waiting for @llamerada-jp 's review before merge.

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by toshipp — APPROVED** [review]

**Pull request merged**
